agent/hcp/client/metrics_client.go

```diff
@@ -144,16 +144,8 @@ func (o *otlpClient) ExportMetrics(ctx context.Context, protoMetrics *metricpb.R
 		return fmt.Errorf("failed to export metrics: %v", err)
 	}
 
-	if respData.Len() != 0 {
```

Comment:
FIX: Check status code and remove useless logic in MetricsClient

When testing the export metrics, I hit a case where I would get a 404: not found because the endpoint wasn't hooked up. This made the code continue and try to deserialize an object that wasn't there. So Added an early return if the status isn't 200: OK.

The `colmetricpb.ExportMetricsServiceResponse` returned is also empty on the Telemetry Gateway side, so don't bother handling it.